backend/app/crud/credentials.py

```
@@ -87,52 +95,92 @@ def get_key_by_org(
+        )
+        raise HTTPException(
+            status_code=404,
+            detail="Credentials not found for this organization and project",
+        )
+
     return creds
 
 
 def get_provider_credential(
     *,
     session: Session,
     org_id: int,
+    project_id: int,
     provider: str,
-    project_id: Optional[int] = None,
     full: bool = False,
-) -> Optional[Union[Dict[str, Any], Credential]]:
+    raise_on_not_found: bool = True,
+) -> dict[str, Any] | Credential | None:
     """
     Fetch credentials for a specific provider within a project.
 
+    Args:
+        session: Database session
+        org_id: Organization ID
+        project_id: Project ID
+        provider: Provider name (e.g., 'openai', 'anthropic')
+        full: If True, returns full Credential object; otherwise returns decrypted dict
+        raise_on_not_found: If True, raises HTTPException when not found; otherwise returns None
+
     Returns:
-        Optional[Union[Dict[str, Any], Credential]]:
+        dict[str, Any] | Credential | None:
             - If `full` is True, returns the full Credential SQLModel object.
             - Otherwise, returns the decrypted credentials as a dictionary.
+            - Returns None if not found and raise_on_not_found is False.
+
+    Raises:
+        HTTPException: If credentials are not found and raise_on_not_found is True
     """
     validate_provider(provider)
 
     statement = select(Credential).where(
         Credential.organization_id == org_id,
         Credential.provider == provider,
-        Credential.is_active == True,
-        Credential.project_id == project_id if project_id is not None else True,
+        Credential.project_id == project_id,
     )
     creds = session.exec(statement).first()
 
     if creds and creds.credential:
         return creds if full else decrypt_credentials(creds.credential)
-    return None
 
+    if not raise_on_not_found:
```

you can avoid creating and maintaining this bool value if you use one_or_none() instead of first()

---

this comment needs attention as well

---

since you have used one or none you would not need to use "raise on not found", but it can be kept this way for now as nothing will break because of this